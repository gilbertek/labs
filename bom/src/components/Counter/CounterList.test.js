import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import CounterList from './CounterList';

describe('CounterList component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<CounterList />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render two counters by default', () => {
    const wrapper = shallow(<CounterList />);
    const counters = wrapper.find('Counter');
    expect(counters.length).toEqual(2);
  });

  it('can add more counters when we click the button', () => {
    const wrapper = shallow(<CounterList />);
    const btn = wrapper.find('button');
    btn.simulate('click');

    const counters = wrapper.find('Counter');
    expect(counters.length).toEqual(3);
  });
});
