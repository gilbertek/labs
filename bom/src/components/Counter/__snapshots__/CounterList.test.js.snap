// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterList component matches the snapshot 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Add Counter
  </button>
  <ul>
    <li>
      <div>
        <p>
          Current count: 
          0
        </p>
        <button
          className="increment"
          onClick={[Function]}
        >
          +++
        </button>
        <button
          className="decrement"
          onClick={[Function]}
        >
          ---
        </button>
      </div>
    </li>
    <li>
      <div>
        <p>
          Current count: 
          0
        </p>
        <button
          className="increment"
          onClick={[Function]}
        >
          +++
        </button>
        <button
          className="decrement"
          onClick={[Function]}
        >
          ---
        </button>
      </div>
    </li>
  </ul>
</div>
`;
